code/datums/components/food/edible.dm

```diff
@@ -214,6 +214,30 @@ Behavior that's still missing from this component that original food items had t
 		var/list/types = bitfield_to_list(foodtypes, FOOD_FLAGS)
 		examine_list += span_notice("It is [lowertext(english_list(types))].")
 
+	if(istype(parent, /obj/item/food))
+		var/obj/item/food/food = parent
+		var/quality = get_preceived_food_quality(user, food)
+		if(quality > 0)
+			var/quality_label = GLOB.food_quality_description[quality]
+			examine_list += span_green("You find this meal [quality_label].")
+		else if (quality == 0)
+			examine_list += span_notice("You find this meal edible.")
+		else
+			examine_list += span_warning("You find this meal inedible.")
+		var/purity = food.reagents.get_average_purity()
+		switch(purity)
+			if(0 to 0.2)
+				examine_list += span_warning("The ingredients are terrible.")
+			if(0.2 to 0.4)
+				examine_list += span_warning("The ingredients are synthetic.")
+			if(0.4 to 0.6)
+				examine_list += span_notice("The ingredients are mediocore.")
+			if(0.6 to 0.8)
+				examine_list += span_green("The ingredients are organic.")
+			if(0.8 to 1)
+				examine_list += span_green("The ingredients are finest.")
```

Comment:
Food quality should not be locked to `/obj/item/food`, it is a template type intended solely for easy application of the edible component.